### PR TITLE
fix(obsidian-nvim): added obsidian.nvim config and changed the readme

### PR DESCRIPTION
**Pull request by mortang2410**
As requested from #309. We need this config to follow links to other markdown files. Please check if anything is amiss. 

Edit: I see the review checklist asks for "Proper usage of `opts` table rather than setting things up with the `config` function." I'm not sure if that is satisfied here.

### CODE REVIEW

File: lua/astrocommunity/note-taking/obsidian-nvim/init.lua [hunk 1]
```diff
@@ -1,17 +1,125 @@
...
+    -- "junegunn/fzf",
+    -- "junegunn/fzf.vim"
+
+    -- Optional, alternative to nvim-treesitter for syntax highlighting.
+    -- "godlygeek/tabular",
+    -- "preservim/vim-markdown",
   },
-  opts = { completion = { nvim_cmp = true } },
+  opts = {
+    dir = "~/obsidian-vault", -- specify the vault location. no need to call 'vim.fn.expand' here
```
Comment (Uzaaft) [inline lua/astrocommunity/note-taking/obsidian-nvim/init.lua:30]: I think it should use `vim.env.HOME `, to make it cross platform. I.e `vim.env.HOME`..`/obsidian-vault`

**Review by Uzaaft — DISMISSED** [review]

**Comment by owittek** [comment]
Wouldn't it be enough to just add your 2 changes to the `init.lua` instead of pasting the entire default config as you're not changing the defaults?

**Comment by mortang2410** [comment]
>  just add your 2 changes

Setting up the `gf` keybinding, for instance, to follow links is done in the config, so we have to copy paste some of it. I also disabled a few things like the `notes`, `daily_notes`, `tabular`, `vim-markdown` (as we already use treesitter), new note ID (it would create new notes upon the slightest mistypes etc.). I think those are sane defaults but please feel free to discuss.

File: lua/astrocommunity/note-taking/obsidian-nvim/init.lua (hunk 1 above)
Comment (mortang2410) [reply lua/astrocommunity/note-taking/obsidian-nvim/init.lua]: Thanks. I made the change.

**Comment by owittek** [comment]
Pardon me then, I came from the issue and the first part as well as the comments seem to be directly taken from the obsidian README.

But my point stands, it's unclear for me what parts deviate from the default config if you also put in all the defaults and leave all the comments in.

Also we'd have to handle line 91-95 in a way that is OS agnostic. For that we'd have to use a function for opts instead.

Lastly we should setup the key bind in a different way than to place it in the `config` function for various reasons.

**Comment by mortang2410** [comment]
> Also we'd have to handle line 91-95 in a way that is OS agnostic. For that we'd have to use a function for opts instead.
> Lastly we should setup the key bind in a different way than to place it in the `config` function for various reasons.

I agree with both of these points. I admit, however, that I am unfamiliar with how astronvim should handle both of these. If you have solutions for them, please share.

**Comment by owittek** [comment]
I can refactor `opts` when I got time and with the key bind part perhaps @RayJameson could help since he did that kind of stuff for harpoon before.

**Comment by RayJameson** [comment]
Shouldn't it work with `keys` property in the plugin setup? I did that thing for Harpoon because there is a top-level mapping that shows in which-key so It would look pretty, otherwise, I would stick with using `keys`

**Comment by owittek** [comment]
Hmm there's still an issue with the setup handler in my code. Testing right now...

**Review by owittek — APPROVED** [review]
I like my own changes, soo ...

File: lua/astrocommunity/note-taking/obsidian-nvim/init.lua [hunk 2]
```diff
@@ -1,17 +1,62 @@
...
+
+    -- Optional, by default when you use `:ObsidianFollowLink` on a link to an external
+    -- URL it will be ignored but you can customize this behavior here.
+    follow_url_func = function(url)
+      local this_os = vim.loop.os_uname().sysname
+      -- Open the URL in the default web browser.
+      if this_os == "Darwin" then
+        vim.fn.jobstart { "open", url }
+      elseif this_os == "Linux" then
+        vim.fn.jobstart { "xdg-open", url }
```
Comment (Uzaaft) [inline lua/astrocommunity/note-taking/obsidian-nvim/init.lua:58]:
```suggestion
        vim.fn.jobstart { "xdg-open", url }
     elseif this_os == "Windows" then
     vim.fn.jobstart { "explorer", url }
     
```

**Comment by Uzaaft** [comment]
@owittek Added a suggestion for windows support as well.

File: lua/astrocommunity/note-taking/obsidian-nvim/init.lua [hunk 3]
```diff
@@ -1,17 +1,65 @@
...
+     elseif this_os == "Windows" then
+     vim.fn.jobstart { "explorer", url }
+
```
Comment (owittek) [inline lua/astrocommunity/note-taking/obsidian-nvim/init.lua:61]:
```suggestion
```

**Review by owittek — COMMENTED** [review]
the pipeline does not like us

**Comment by Uzaaft** [comment]
@owittek  Apparently `utils.system_open` should be enough

File: lua/astrocommunity/note-taking/obsidian-nvim/init.lua [hunk 4]
```diff
@@ -1,17 +1,64 @@
...
+      local this_os = vim.loop.os_uname().sysname
+      -- Open the URL in the default web browser.
+      if this_os == "Darwin" then
+        vim.fn.jobstart { "open", url }
+      elseif this_os == "Linux" then
+        vim.fn.jobstart { "xdg-open", url }
+     elseif this_os == "Windows" then
+     vim.fn.jobstart { "explorer", url }
+      end
+    end,
```
Comment (mehalter) [inline lua/astrocommunity/note-taking/obsidian-nvim/init.lua:62]:
```suggestion
    follow_url_func = vim.ui.open or require("astronvim.utils").system_open,
```

Comment (Uzaaft) [inline lua/astrocommunity/note-taking/obsidian-nvim/init.lua:62]:
```suggestion
    follow_url_func = require("astronvim.utils").system_open
```

**Review by mehalter — CHANGES_REQUESTED** [review]
@owittek

**Review by Uzaaft — APPROVED** [review]

**Review by mehalter — APPROVED** [review]
LGTM

**Pull request merged**